src/core/timesync.cpp

```diff
@@ -75,6 +75,7 @@ void Timesync::set_timesync_offset(int64_t offset_ns, uint64_t start_transfer_lo
 
         // Save time offset for other components to use
         _parent.get_autopilot_time().shift_time_by(std::chrono::nanoseconds(offset_ns));
+        _timesync_locked = true;
```

Comment:
May be need to unlock timesync if we have no response?

```
if (_high_rtt_count > _MAX_CONS_HIGH_RTT) {
    // Issue a warning to the user if the RTT is constantly high
    LogWarn() << "RTT too high for timesync: " << rtt_ns / 1000000.0 << " ms.";
    // Reset counter
     _high_rtt_count = 0;
     _timesync_locked = false;
}
```
https://github.com/mavlink/MAVSDK/pull/1034/files#diff-ec4205b6bc1adbc844894d870f234adaL85-L91

Reply:
If we only have high RTT, I'm not sure what the best case is. I'd prefer to not confuse the flight controller if the clock on MAVSDK isn't stable...

Reply:
I worry about case of MAVSDK will reconnect to FC during one session. Sometimes it can be a different FCs or FC with changed behaviour.

Reply:
So may be in this place we should reset a flag:
```C++
 if (_parent.is_connected()) {
    uint64_t now_ns = std::chrono::duration_cast<std::chrono::nanoseconds>(
                          _parent.get_autopilot_time().now().time_since_epoch())
                          .count();
    send_timesync(0, now_ns);
} else {
    _autopilot_timesync_acquired = false;
}
```
https://github.com/mavlink/MAVSDK/blob/899ff7ab830cbcb404a14f21164ae991393b1a78/src/core/timesync.cpp#L22-L33

Reply:
Ok yes, good idea. I've added that.